File: packages/cpp/tests/dxf_export_test.cpp
#include "openskp/dxf_export.hpp"

#include <gtest/gtest.h>

namespace openskp {
namespace {

TEST(DxfExport, SerializesSceneTo3DxfText) {
  Scene scene;
  GlbPrimitive prim;
  prim.geom_name = "Walls";
  prim.material_index = 0;
  prim.positions = {0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0};
  prim.normals = {0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0};
  prim.uvs = {0.0, 0.0, 1.0, 0.0, 0.0, 1.0};
  prim.indices = {0, 1, 2};
  scene.glb_primitives.push_back(prim);

  std::string dxf_text = to_dxf(scene);
  EXPECT_NE(dxf_text.find("$ACADVER"), std::string::npos);
  EXPECT_NE(dxf_text.find("AC1015"), std::string::npos);
  EXPECT_NE(dxf_text.find("POLYLINE"), std::string::npos);
  EXPECT_NE(dxf_text.find("AcDbPolyFaceMesh"), std::string::npos);
  EXPECT_NE(dxf_text.find("Walls"), std::string::npos);
  EXPECT_NE(dxf_text.find("EOF"), std::string::npos);

  std::string dxf_3d = to_dxf(scene, METRES_TO_INCHES, "3dface");
  EXPECT_NE(dxf_3d.find("3DFACE"), std::string::npos);
  EXPECT_NE(dxf_3d.find("AcDbFace"), std::string::npos);
}

}  // namespace
}  // namespace openskp

File: packages/dart/test/dxf_export_test.dart
import 'dart:typed_data';
import 'package:openskp/openskp.dart';
import 'package:test/test.dart';

void main() {
  group('DXF 3D Exporter', () {
    test('serializes Scene to 3D DXF text format', () {
      final scene = Scene(
        sceneHierarchy: InstanceNode(
          name: 'Root',
          definitionName: 'RootDef',
          layer: 'Layer0',
          positionMm: (0.0, 0.0, 0.0),
          properties: {},
          children: [],
        ),
        meshIndex: {},
        glbPrimitives: [
          GlbPrimitive(
            geomName: 'Walls',
            materialIndex: 0,
            positions: Float32List.fromList([0, 0, 0, 1, 0, 0, 0, 1, 0]),
            normals: Float32List.fromList([0, 0, 1, 0, 0, 1, 0, 0, 1]),
            uvs: Float32List.fromList([0, 0, 1, 0, 0, 1]),
            indices: Uint32List.fromList([0, 1, 2]),
          ),
        ],
        gltfMaterials: [],
      );

      final dxfText = toDxf(scene);
      expect(dxfText, contains('\$ACADVER'));
      expect(dxfText, contains('AC1015'));
      expect(dxfText, contains('POLYLINE'));
      expect(dxfText, contains('AcDbPolyFaceMesh'));
      expect(dxfText, contains('Walls'));
      expect(dxfText, contains('EOF'));

      final dxf3d = toDxf(scene, mode: '3dface');
      expect(dxf3d, contains('3DFACE'));
      expect(dxf3d, contains('AcDbFace'));
    });
  });
}
